docs/reference/filebeat/configuration-autodiscover-hints.md

```diff
@@ -146,15 +152,23 @@ filebeat.autodiscover.providers:
   - type: docker
     hints.enabled: true
     hints.default_config:
-      type: container
+      type: filestream
+      id: container-${data.container.id}
+      prospector.scanner.symlinks: true
+      parsers:
+        - container: ~
       paths:
         - /var/lib/docker/containers/${data.container.id}/*.log
     templates:
       - condition:
           equals:
             docker.container.labels.type: "pipeline"
         config:
-          - type: container
+          - type: filestream
+            id: container-${data.container.id}
```

Comment:
That should match what is in the `paths`: `${data.docker.container.id}`

Honestly, I'm always confused if the correct prefix is `data.docker` or just `data`. The original code uses a mix of both :man_facepalming: 

It might be a difference between hints/no hints.

I highly recommend testing each of those configurations before merging this PR to ensure they actually work with our `main` branch or at least 9.0.

Reply:
Yes, I'm planning to test these. Thank for catching this.

Reply:
Let me know once it's ready for final review.

Reply:
I've tested changed code snippets and both `${data.container.id}` and `${data.docker.container.id}`. Looks both variables get populated with the same data. My theory is that in case of a different container engine we have a more generalized `container.id` that is not Docker-specific.

I'm merging this PR now.